pkgs/top-level/python-packages.nix

```diff
@@ -1887,6 +1887,8 @@ in {
 
   pkginfo = callPackage ../development/python-modules/pkginfo { };
 
+  pre-commit = disabledIf isPy27 (toPythonModule (pkgs.gitAndTools.pre-commit));
```

Comment:
I'm seeing support here for python 2.7 https://github.com/pre-commit/pre-commit/blob/v1.18.0/setup.cfg#L38

Reply:
interesting... I was basing that off the gitAndTools.pre-commit package.....

Reply:
I remember now, the application uses python3Packages, I can invert the logic..... and move pre-commit to being a python package, and then just do `toPythonApplication python3Packages.pre-commit` in the all-packages.nix.

But.... seeing as I don't have a reason to use it in this PR, i might remove the related changes altogether.